WooCommerce/Classes/ViewRelated/Products/ShareProduct/ShareProductCoordinator.swift

```diff
@@ -91,13 +91,13 @@ private extension ShareProductCoordinator {
             DDLogWarn("⚠️ No site found for generating product sharing message!")
             return
         }
-        let viewModel = ProductSharingMessageGenerationViewModel(siteID: siteID, productName: productName, url: productURL.absoluteString)
-        let controller = ProductSharingMessageGenerationHostingController(viewModel: viewModel) { [weak self] message in
-            self?.navigationController.topmostPresentedViewController.dismiss(animated: true) {
-                self?.presentShareSheet(with: message)
-            }
+        let viewModel = ProductSharingMessageGenerationViewModel(siteID: siteID,
+                                                                 productName: productName,
+                                                                 url: productURL.absoluteString,
+                                                                 onShare: {
```

Comment:
Nit: if we are keeping this closure just for analytics, please consider removing it for simplicity because we'll need to add analytics in the view model to track the generation completion/failure anyway.

Reply:
I agree. Removed the closure in https://github.com/woocommerce/woocommerce-ios/pull/9901/commits/757c69b64f6897d05bda946fa5e5534d32c985fb